src/ReservoirPair.sol

```diff
@@ -385,6 +385,13 @@ abstract contract ReservoirPair is IAssetManagedPair, ReservoirERC20 {
         }
     }
 
+    function _skimExcessManaged(ERC20 aToken, uint256 aAmtManaged) internal returns (uint256 rAmtSkimmed) {
+        address lRecoverer = factory.read(RECOVERER_NAME).toAddress();
+        rAmtSkimmed = aAmtManaged - type(uint104).max;
+        assetManager.returnAsset(aToken == _token0(), rAmtSkimmed);
+        _safeTransfer(address(aToken), lRecoverer, rAmtSkimmed);
```

Comment:
I noticed, we should replace uni v2's impl of `_safeTransfer` with a lib implementation, yeah?

Reply:
Yep, sounds good

Reply:
okay for this I realize I cannot replace all instances of it. For this particular instance I can. 

But for `_checkedTransfer`, we need to get a return `bool` and not revert, which the library does not do. `try/catch` also doesn't work with the lib cuz it's assembly and not recognized as an external function call. So it seems that we have to keep our `_safeTransfer` for now

Reply:
Try catch not working is not because its assembly, its because its an internal call. _safeTransfer does not use `CALL` it uses `JUMP`. Once it jumps to the internal function, that function then uses `CALL`, so you cannot catch it from a layer up sadly.

Reply:
Mixing different types of _safeTransfer scares me a little. Is it only inside of checkedTransfer that we use _safeTransfer? If so, can we just inline the call and delete _safeTransfer?

Reply:
yes it's only `_checkedTransfer` that uses `_safeTransfer` now. But there's **two** instances so if we inline we'll be repeating code?

Reply:
Okay, lets not inline